File: src/main/java/hu/unideb/inf/WebProject/controller/PerformerController.java
package hu.unideb.inf.WebProject.controller;

import hu.unideb.inf.WebProject.DTO.PerformerDTO;
import hu.unideb.inf.WebProject.service.PerformerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "http://localhost:3000/")
@RequestMapping("/performers")
public class PerformerController {

    @Autowired
    PerformerService performerService;

    @PostMapping("/createPerformer")
    ResponseEntity createPerformer(@RequestBody PerformerDTO performerDTO) {
        return new ResponseEntity(performerService.create(performerDTO), HttpStatus.CREATED);
    }

    @GetMapping("/getAll")
    ResponseEntity getAllPerformers() {
        return new ResponseEntity<>(performerService.readAll(), HttpStatus.OK);
    }

    @PatchMapping("/updatePerformer")
    ResponseEntity updatePerformer(@RequestBody PerformerDTO performerDTO) {
        return new ResponseEntity(performerService.update(performerDTO), HttpStatus.OK);
    }

    @DeleteMapping("/deletePerformer")
    ResponseEntity deletePerformer(@RequestBody PerformerDTO performerDTO) {
        performerService.delete(performerDTO.performerId());
        return new ResponseEntity(HttpStatus.OK);
    }
}

File: frontend/src/performerController.js
import axios from "axios"

export class PerformerController {
    static async createPerformer(performerDTO){
        const respons = await axios.post('http://localhost:8080/rqimmp/performers/createPerformer', performerDTO)
        return respons.data
    }

    static async getAll() {
        const respons = await axios.get('http://localhost:8080/rqimmp/performers/getAll')
        return respons.data
    }

    static async updatePerformer(performerDTO) {
        const respons = await axios.patch('http://localhost:8080/rqimmp/performers/updatePerformer', performerDTO)
        return respons.data
    }

    static async deletePerformer(performerDTO) {
        const respons = await axios.delete('http://localhost:8080/rqimmp/performers/deletePerformer' , {data : performerDTO})
        return respons.data
    }
}